bson/src/test/unit/org/bson/types/ObjectIdTest.java

```diff
@@ -146,33 +147,32 @@ public void testToHexString() {
                 new ObjectId(Integer.MAX_VALUE, Short.MAX_VALUE, Short.MAX_VALUE, Short.MAX_VALUE).toHexString());
     }
 
-    private void checkDates(final String dateInTime, final ObjectId oid) {
-        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm:ssZ");
-        try {
-            Date d = dateFormat.parse(dateInTime);
-            assertEquals(oid.getDate(), d);
-        } catch (Exception e) {
-        }
-    }
-
     @Test
-    public void testTimeZero() {
-        checkDates("01-Jan-1970 00:00:00", new ObjectId(0, 0));
+    public void testTimeZero() throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss Z");
+        Date d = dateFormat.parse("01-Jan-1970 00:00:00 -0000");
+        assertEquals(new ObjectId(0, 0).getDate(), d);
```

Comment:
Reverse parameters in all the methods: the first parameter to assertEquals is the expected value, not the second (it matters when the test fails, and Junit reports the error.  it's confusing when the expected value is the actual value)

Reply:
Done